### PR TITLE
Rename transactionHash property to hash in the persisted transaction metadata

### PR DESCRIPTION
**Pull request by vinistevam**
## Explanation
Rename the `transactionHash` property to `hash` in the persisted transaction metadata as part of the unification initiative between the core and extension. 

The following task will take place to handle the migration and adoption of the patch/release.

## References

Fixes https://github.com/MetaMask/MetaMask-planning/issues/1077

### `@metamask/transaction-controller`

- **BREAKING**:  rename the `transactionHash` property to `hash` in the persisted `TransactionMeta`.

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/EtherscanRemoteTransactionSource.test.ts [hunk 1]
```diff
@@ -102,7 +102,7 @@ const EXPECTED_NORMALISED_TRANSACTION_BASE = {
     to: ETHERSCAN_TRANSACTION_SUCCESS_MOCK.to,
     value: '0xb1a2bc2ec50000',
   },
-  transactionHash: ETHERSCAN_TRANSACTION_SUCCESS_MOCK.hash,
+  hash: ETHERSCAN_TRANSACTION_SUCCESS_MOCK.hash,
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/EtherscanRemoteTransactionSource.test.ts:105]: Very minor but should we move this to maintain alphabetical order?

File: packages/transaction-controller/src/EtherscanRemoteTransactionSource.ts [hunk 2]
```diff
@@ -154,7 +154,7 @@ export class EtherscanRemoteTransactionSource
         to: txMeta.to,
         value: BNToHex(new BN(txMeta.value)),
       },
-      transactionHash: txMeta.hash,
+      hash: txMeta.hash,
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/EtherscanRemoteTransactionSource.ts:157]: Alphabetical order?

File: packages/transaction-controller/src/EtherscanRemoteTransactionSource.test.ts (hunk 1 above)
Comment (matthewwalsh0) [reply packages/transaction-controller/src/EtherscanRemoteTransactionSource.test.ts:105]: Same applies to most usages of the new property.

**Comment by vinistevam** [comment]
> Same question as for [#1624 (comment)](https://github.com/MetaMask/core/pull/1624#issuecomment-1693026551)

Thanks again for pointing it out. :pray: 
It was mainly motivated to reduce risk as the extension is already using `hash` and the changes necessary on mobile would be just doing the migration and renaming in a few places on the other hand in the extension it would be a much bigger change.

File: packages/transaction-controller/src/EtherscanRemoteTransactionSource.test.ts (hunk 1 above)
Comment (vinistevam) [reply packages/transaction-controller/src/EtherscanRemoteTransactionSource.test.ts]: Sure, I just pushed the changes. :pray:

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Pull request merged**